Sources/Support/PurchasesDiagnostics.swift

```diff
@@ -59,36 +117,74 @@ extension PurchasesDiagnostics {
         /// Failure performing a signed request
         case failedMakingSignedRequest(Swift.Error)
 
+        /// Version of the SDK is not supported
+        case invalidSDKVersion
+
+        /// There are no offerings in project
+        case noOfferings
+
+        /// Offerings are not configured correctly
+        case offeringConfiguration([OfferingConfigurationErrorPayload])
+
+        /// App bundle ID does not match the one set in the dashboard
+        case invalidBundleId(InvalidBundleIdErrorPayload?)
+
+        /// One or more products are not configured correctly
+        case invalidProducts([InvalidProductErrorPayload])
+
+        /// The person is not authorized to make In-App Purchases
+        case notAuthorizedToMakePayments
+
         /// Any other not identifier error. You can check the undelying error for details.
         case unknown(Swift.Error)
 
     }
 
 }
 
-// MARK: - Implementation
-
 extension PurchasesDiagnostics {
+    /// Status of the SDK Health report
+    public enum SDKHealthStatus {
+        /// SDK configuration is valid but might have some non-blocking issues
+        case healthy(warnings: [PurchasesDiagnostics.Error])
+        /// SDK configuration is not valid and has issues that must be resolved
+        case unhealthy(PurchasesDiagnostics.Error)
+    }
 
-    /// Perform tests to ensure SDK is configured correctly.
-    /// - `Throws`: ``PurchasesDiagnostics/Error`` if any step fails
-    @objc(testSDKHealthWithCompletion:)
+    /// Throw an error if the SDK is not configured correctly
     public func testSDKHealth() async throws {
-        do {
-            try await self.unauthenticatedRequest()
-            #if !ENABLE_CUSTOM_ENTITLEMENT_COMPUTATION
-            try await self.authenticatedRequest()
-            #endif
-            try await self.offeringsRequest()
-            try await self.signatureVerification()
-        } catch let error as Error {
-            throw error
-        } catch let error {
-            // Catch every other error to ensure that we only throw `Error`s from here.
-            throw Error.unknown(error)
+        switch await self.healthReport() {
+        case let .unhealthy(error): throw error
+        default: break
         }
     }
 
+    private var canMakePayments: Bool {
+        if #available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, visionOS 1.0, *) {
+            return AppStore.canMakePayments
+        } else {
+            return SKPaymentQueue.canMakePayments()
+        }
+    }
```

Comment:
we already have our own API available through Purchases, let's use that instead

Reply:
also I'm now realizing that we haven't updated it for SK2 (cc @fire-at-will @MarkVillacampa) so we might as well add replace that method's implementation with this one lol 
https://github.com/RevenueCat/purchases-ios/blob/d9626ab8574c88f1bbde7e2610e2a7adcebfde48/Sources/Purchasing/Purchases/Purchases.swift#L162

(can be a separate PR, this isn't exactly urgent or blocking)

Reply:
Oh that's a very good point! Let's definitely fix it in a follow-up PR 🚀 💪